storage/storage.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Adapted from istio/proxy/test/backend/echo with error handling and
+// concurrency fixes and making it as low overhead as possible
+// (no std output by default)
+
+package storage // import "istio.io/fortio/storage"
+
+import (
+	"time"
+
+	"istio.io/fortio/log"
+
+	"cloud.google.com/go/datastore"
```

Comment:
so far I have no external dependencies outside of grpc (and the grpc part is optional)

I wonder if we should put that gcp specific part somewhere else (istio/istio/tools/perf or istio/test-infra...)

Reply:
There is. But will be ugly in code (we need to somehow manually grab credentials, and manually construct a giant JSON).

Reply:
Can be done somewhat easier with a shell + python script though.